src/treq/client.py

```diff
@@ -431,6 +448,29 @@ def _guess_content_type(filename):
     return guessed or 'application/octet-stream'
 
 
+def _get_content_type_from_header(headers, default=None):
+    """
+    Retrieve Content-Type header from twisted.web.http_headers.Headers object
+    or return supplied default
+
+    :param headers:
+        A twisted.web.http_headers.Headers object
+
+    :param default:
+        The default Content-Type to return, encoded as a byte-string
+
+    :returns:
+        byte-string containing a Content-Type value
+    :rtype:
+        bytes
+    """
+    if not headers:
+        return default
+    return headers.getRawHeaders(
```

Comment:
I think that this function should raise an exception when there are multiple content-type headers (since that's not valid). I think it's okay to break messily in this case, so you could unpack like:

```py
[content_type] = headers.getRawHeaders(CONTENT_TYPE_HEADER_NAME, [default])
return content_type
```